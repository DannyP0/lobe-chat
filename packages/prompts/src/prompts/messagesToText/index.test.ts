import { describe, expect, it } from 'vitest';

import { Message, conversationToText } from './index';

describe('conversationToText', () => {
  it('should format a single message correctly', () => {
    const messages: Message[] = [{ role: 'user', content: 'Hello, how are you?' }];

    const result = conversationToText(messages);

    expect(result).toMatchSnapshot();
  });

  it('should format multiple messages correctly', () => {
    const messages: Message[] = [
      { role: 'user', content: 'Hello, how are you?' },
      { role: 'assistant', content: 'I am doing well, thank you!' },
      { role: 'user', content: 'What is the weather like?' },
    ];

    const result = conversationToText(messages);

    expect(result).toMatchSnapshot();
  });

  it('should handle empty messages array', () => {
    const messages: Message[] = [];

    const result = conversationToText(messages);

    expect(result).toMatchSnapshot();
  });

  it('should handle messages with different roles', () => {
    const messages: Message[] = [
      { role: 'system', content: 'You are a helpful assistant.' },
      { role: 'user', content: 'Tell me a joke.' },
      { role: 'assistant', content: 'Why did the chicken cross the road?' },
    ];

    const result = conversationToText(messages);

    expect(result).toMatchSnapshot();
  });

  it('should handle messages with special characters', () => {
    const messages: Message[] = [
      { role: 'user', content: 'What is 2 + 2?' },
      { role: 'assistant', content: 'The answer is 4.' },
    ];

    const result = conversationToText(messages);

    expect(result).toMatchSnapshot();
  });

  it('should handle multiline content', () => {
    const messages: Message[] = [
      {
        role: 'user',
        content: `This is a multiline message.
It has multiple lines.
Line 3 here.`,
      },
    ];

    const result = conversationToText(messages);

    expect(result).toMatchSnapshot();
  });
});
