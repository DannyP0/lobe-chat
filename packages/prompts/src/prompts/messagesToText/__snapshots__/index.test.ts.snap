// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`conversationToText > should format a single message correctly 1`] = `
"<conversation>
<user>Hello, how are you?</user>
</conversation>"
`;

exports[`conversationToText > should format multiple messages correctly 1`] = `
"<conversation>
<user>Hello, how are you?</user>
<assistant>I am doing well, thank you!</assistant>
<user>What is the weather like?</user>
</conversation>"
`;

exports[`conversationToText > should handle empty messages array 1`] = `"<conversation/>"`;

exports[`conversationToText > should handle messages with different roles 1`] = `
"<conversation>
<system>You are a helpful assistant.</system>
<user>Tell me a joke.</user>
<assistant>Why did the chicken cross the road?</assistant>
</conversation>"
`;

exports[`conversationToText > should handle messages with special characters 1`] = `
"<conversation>
<user>What is 2 + 2?</user>
<assistant>The answer is 4.</assistant>
</conversation>"
`;

exports[`conversationToText > should handle multiline content 1`] = `
"<conversation>
<user>This is a multiline message.
It has multiple lines.
Line 3 here.</user>
</conversation>"
`;
